src/mockServiceWorker.js

```diff
@@ -321,8 +319,8 @@ function sendToClient(client, message, body) {
     }
 
     const transfer = [channel.port2]
-    if (body) {
-      transfer.push(body)
+    if (typeof message.payload !== 'undefined' && message.payload.body) {
```

Comment:
What do you think if we flip the responsibility around? Instead of `sendToClient` checking a potential body input, we'd expose a callback that allows the consumer to push additional data to the transport?

```js
const clientMessage = await sendToClient(
  client,
  requestObj,
  (transport) => transport.push(request.body)
)

// ...

function sendToClient(client, message, goodNameHere) {
  if (goodNameHere) {
    goodNameHere(transport)
  }
}
```

The only question now is whether we can call `goodNameHere` at a certain point of the transport being constructed. I don't see why we wouldn't be able to, but I feel you have better understanding of whether this suggestion makes sense.

Reply:
I think it makes it more or less the same as my original approach.

Reply:
Then I'd move the responsibility of pushing to the `transport` to the consumer of `sentToClient`. Imo, it makes the function more predictable, as it doesn't try to leverage the complexity of input, delegating that complexity to the consumer (closer to the context).

Reply:
SendToClient is the last one on this side of the channel. I think it make the most sense at callers to provide transferrable content. We can make it similar to post message - an array of transferrable content. Others will just pass empty array